cross-chain/solana/solana-tbtc-anchor/tests/solana-tbtc-anchor.ts

```diff
@@ -0,0 +1,497 @@
+    } catch (_err) {
+      expect(_err).to.be.instanceOf(AnchorError);
+      const err: AnchorError = _err;
+      expect(err.error.errorCode.code).to.equal('ConstraintSeeds');
+      expect(err.program.equals(program.programId)).is.true;
+    }
+  });
+
+  it('use two minters', async () => {
+    const tbtcKeys = anchor.web3.Keypair.generate();
+    await setup(program, tbtcKeys, creator);
+    await checkState(program, tbtcKeys, creator, 0, 0, 0);
+    const minterInfoPDA = await addMinter(program, tbtcKeys, creator, minterKeys, creator);
+    const minter2InfoPDA = await addMinter(program, tbtcKeys, creator, minter2Keys, creator);
+    await checkMinter(program, tbtcKeys, minterKeys);
+    await checkMinter(program, tbtcKeys, minter2Keys);
+    await checkState(program, tbtcKeys, creator, 2, 0, 0);
+    await setupMint(program, tbtcKeys, creator, recipientKeys);
+
+    // cannot mint with wrong keys
+    try {
+      await mint(program, tbtcKeys, minter2Keys, minterInfoPDA, recipientKeys, 1000);
+      chai.assert(false, "should've failed but didn't");
+    } catch (_err) {
+      expect(_err).to.be.instanceOf(AnchorError);
+      const err: AnchorError = _err;
+      expect(err.error.errorCode.code).to.equal('ConstraintSeeds');
+      expect(err.program.equals(program.programId)).is.true;
+    }
+
+    await mint(program, tbtcKeys, minterKeys, minterInfoPDA, recipientKeys, 500);
+    await checkState(program, tbtcKeys, creator, 2, 0, 500);
+  });
+
+  it('remove minter', async () => {
+    const tbtcKeys = anchor.web3.Keypair.generate();
+    await setup(program, tbtcKeys, creator);
+    await checkState(program, tbtcKeys, creator, 0, 0, 0);
+    const minterInfoPDA = await addMinter(program, tbtcKeys, creator, minterKeys, creator);
+    await checkMinter(program, tbtcKeys, minterKeys);
+    await checkState(program, tbtcKeys, creator, 1, 0, 0);
+    await removeMinter(program, tbtcKeys, creator, minterKeys, minterInfoPDA);
+    await checkState(program, tbtcKeys, creator, 0, 0, 0);
+  });
+
+  it('won\'t remove minter', async () => {
+    const tbtcKeys = anchor.web3.Keypair.generate();
+    await setup(program, tbtcKeys, creator);
+    await checkState(program, tbtcKeys, creator, 0, 0, 0);
+    const minterInfoPDA = await addMinter(program, tbtcKeys, creator, minterKeys, creator);
+    await checkMinter(program, tbtcKeys, minterKeys);
+    await checkState(program, tbtcKeys, creator, 1, 0, 0);
+    await removeMinter(program, tbtcKeys, creator, minterKeys, minterInfoPDA);
+    await checkState(program, tbtcKeys, creator, 0, 0, 0);
+
+    try {
+      await removeMinter(program, tbtcKeys, creator, minterKeys, minterInfoPDA);
+    } catch (_err) {
+      expect(_err).to.be.instanceOf(AnchorError);
+      const err: AnchorError = _err;
+      expect(err.error.errorCode.code).to.equal('AccountNotInitialized');
+      expect(err.program.equals(program.programId)).is.true;
+    }
+  });
+
+  it('add guardian', async () => {
```

Comment:
I think it would be nice having tests that try to add/remove guardians with the wrong keys.